packages/x-components/jest.setup.ts

```diff
@@ -1 +1,7 @@
 import '@testing-library/jest-dom/extend-expect';
+
+window.ResizeObserver = jest.fn().mockImplementation(() => ({
```

Comment:
I'm missing some tests for this behavior. Can we move the current `base-grid` test to Cypress component testing? This component is now pretty coupled to the view because of the resizeobserver, so I don't think it makes sense to keep using `jest` for it. If you find out it is too complicated for this task, we can do it in another sub-task.